provider/provider_test.go

```diff
@@ -87,3 +125,678 @@ func TestProviderCdpRegion(t *testing.T) {
+				envValue: "false",
+			},
+			modelValue:     types.BoolNull(),
+			expectedResult: false,
+		},
+
+		{
+			base: cdpConfigTestScenarioBase{
+				name:     "test when LocalEnvironment is null and env var value is true",
+				envVar:   "LOCAL_ENVIRONMENT",
+				envValue: "true",
+			},
+			modelValue:     types.BoolNull(),
+			expectedResult: true,
+		},
+
+		{
+			base: cdpConfigTestScenarioBase{
+				name:     "test when LocalEnvironment is null and env var value is empty",
+				envVar:   "LOCAL_ENVIRONMENT",
+				envValue: "",
+			},
+			modelValue:     types.BoolNull(),
+			expectedResult: false,
+		},
+	}
+	for _, test := range tests {
+		t.Setenv(test.base.envVar, test.base.envValue)
+		t.Run(test.base.name, func(t *testing.T) {
+			model := createCdpProviderModel()
+			model.LocalEnvironment = test.modelValue
+
+			config := getCdpConfig(context.Background(), model, testVersion, testTerraformVersion)
+
+			if config.LocalEnvironment != test.expectedResult {
+				t.Errorf("LocalEnvironment diverges from the expected! Expected: %t, got: %t\n",
+					test.expectedResult, config.LocalEnvironment)
+			}
+		})
+		t.Setenv(test.base.envVar, "")
+	}
+}
+
+func TestGetOrDefaultFromEnvWithValue(t *testing.T) {
+	val := types.StringValue("expected")
+	if got := getOrDefaultFromEnv(val, testEnvVariable); got != val.ValueString() {
+		t.Errorf("Assertion faled! Expected: %s, got: %s\n", val.ValueString(), got)
+	}
+}
+
+func TestGetOrDefaultFromEnvWhenValuePresentInEnv(t *testing.T) {
+	val := "expected"
+	t.Setenv(testEnvVariable, val)
+	if got := getOrDefaultFromEnv(types.StringNull(), testEnvVariable); got != val {
+		t.Errorf("Assertion faled! Expected: %s, got: %s\n", val, got)
+	}
+}
+
+func TestGetOrDefaultFromEnvWhenValuePresentInEnvButNotFirst(t *testing.T) {
+	val := "expected"
+	t.Setenv(testEnvVariable, val)
+	if got := getOrDefaultFromEnv(types.StringNull(), "customOtherEnvVariable", testEnvVariable); got != val {
+		t.Errorf("Assertion faled! Expected: %s, got: %s\n", val, got)
+	}
+}
+
+func TestGetOrDefaultBoolFromEnvWithValue(t *testing.T) {
+	val := types.BoolValue(true)
+	if got := getOrDefaultBoolFromEnv(context.Background(), val, testEnvVariable); got != val.ValueBool() {
+		t.Errorf("Assertion faled! Expected: %t, got: %t\n", val.ValueBool(), got)
+	}
+}
+
+func TestGetOrDefaultBoolFromEnvWhenValuePresentInEnv(t *testing.T) {
+	val := "true"
+	t.Setenv(testEnvVariable, val)
+	if got := getOrDefaultBoolFromEnv(context.Background(), types.BoolNull(), testEnvVariable); !got {
+		t.Errorf("Assertion faled! Expected: %t, got: %t\n", true, got)
+	}
+}
+
+func TestGetOrDefaultBoolFromEnvWhenValuePresentInEnvButNotFirst(t *testing.T) {
+	val := "true"
+	t.Setenv(testEnvVariable, val)
+	if got := getOrDefaultBoolFromEnv(context.Background(), types.BoolNull(), "customOtherEnvVariable", testEnvVariable); !got {
+		t.Errorf("Assertion faled! Expected: %t, got: %t\n", true, got)
+	}
+}
+
+func TestResources(t *testing.T) {
```

Comment:
I think the underlying functions (provider.Resources) will be implicitly tested for each resource when we add acceptance tests for all resources. Having these tests will not hurt, but I don't think they will be super useful.

Reply:
Yeah, it was kind of an overkill but I wanted to ensure that modifying this list of resources cannot be done unintentionally. For example during development/testing, it can be modified - like temporarily commenting out some of the resources - but in the end, this test shall ensure that this temporary change won't affect anything later on and cannot/shouldn't pass the code review (I have faith in ourselves but accidents happens when folks carrying heavy load)